### ISSUE TITLE
Rigid Bodies floating in the air

### ISSUE THREAD

**Issue by jonri** [godotengine/godot#29906]
**Godot version:**
First appeared in https://github.com/godotengine/godot/commit/a9d4cde0f5abc399332dde0c7415b99a5796eb34 (result of git bisect)

**OS/device including version:**
Linux

**Issue description:**

A project that worked as expected in Godot 3.1.1 Stable has issues with rigid bodies in the latest master.

Actual result: rigid bodies float in the air, remain there when hit, or overlap each other.
![physics-bad](https://user-images.githubusercontent.com/6228151/59785489-e5cfeb80-9292-11e9-9c45-ca9cd56acf99.png)

Expected result: rigid bodies fall to ground
![physics-good](https://user-images.githubusercontent.com/6228151/59785490-e5cfeb80-9292-11e9-94cc-12ddb76611ed.png)

The commit in question added the following lines:
https://github.com/godotengine/godot/commit/a9d4cde0f5abc399332dde0c7415b99a5796eb34#diff-51c3f48359856e1f53cfeb65f82a859fR302

On its own this is correct, as it prevents an unchanged property from firing a spurious shape_changed().  However, taking out the guard fixes the issue.  This leads me to believe that some other function should be calling shape_changed() and is not, and the fact that it used to be always called here was covering for it.

**Steps to reproduce:**
1.  Run the project below, turn left to find some floating rigid bodies.
2.  Comment out the two lines listed above and rebuild the engine.
3.  Re-run the project and the rigid bodies will work normally.

**Minimal reproduction project:**
Not minimal, but easily reproduced in my project: https://github.com/jonri/StuntPlaygroundGodot

**Comment by akien-mga** [comment]
CC @ShyRed @AndreaCatania

**Comment by Ace-Dragon** [comment]
If you reset the scale of the collision shapes to 1 and instead resize them using the handles, then the rigid bodies should touch the ground as they should.

I'm not sure if the change is intended though.

**Comment by AndreaCatania** [comment]
Do you have the sleeping on? if Yes this is the reason

**Comment by jonri** [comment]
When I imported the original collision objects into Blender, I failed to apply the scale on a couple of them before re-exporting.  The problem is indeed limited only to the ones that have a scale on the shape.

I'll definitely fix my shapes since we consider it bad practice to scale collision shapes, but it was actually working in the previous release so we might still want to consider making the code work again.  Given that it worked with the extra shape_changed() call, it might just be a matter of calling that when the scale is set.

If we want to support this, I can try to figure it out.  Or if the official position is that we shouldn't support it, I think we should have an editor warning flag show up when the scale is set.
